server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -440,10 +440,12 @@ public TextSearchInfo getTextSearchInfo() {
      * @param caseInsensitive if matches should be case insensitive
      * @param string the partially complete word the user has typed (can be empty)
      * @param queryShardContext the shard context
+     * @param searchAfter - usually null. If supplied the TermsEnum result must be positioned after the provided term (used for pagination)
      * @return null or an enumeration of matching terms and their doc frequencies
      * @throws IOException Errors accessing data
      */
-    public TermsEnum getTerms(boolean caseInsensitive, String string, SearchExecutionContext queryShardContext) throws IOException {
+    public TermsEnum getTerms(boolean caseInsensitive, String string, SearchExecutionContext queryShardContext, BytesRef searchAfter)
+        throws IOException {
         return null;
```

Comment:
We have a whole bunch of methods on MappedFieldType which deal only with text search, I wonder if we should move some of these into TextSearchInfo instead? One to put on the refactoring list...

Reply:
Given that the input string is a String, maybe the searchAfter parameter should be one too?